src/test/java/io/vertx/jdbcclient/JDBCPoolInitTest.java

```diff
@@ -107,4 +117,55 @@ public void test_init_pool_by_existing_datasource_custom_config(TestContext shou
     client = JDBCPool.pool(vertx, ds, config);
     simpleAssertSuccess(should);
   }
+
+  @Test
+  public void test_init_pool_with_datasource_name(TestContext should) {
+    final String datasourceName = "customDatasourceName";
+
+    final MetricsOptions metricsOptions = new MetricsOptions().setEnabled(true);
+    metricsOptions.setFactory(vertxOptions -> new VertxMetrics() {
+      @Override
+      public PoolMetrics<?> createPoolMetrics(String poolType, String poolName, int maxPoolSize) {
```

Comment:
we can move this code to the `setUp` method.

Then instead of asserting here, when the pool is created we capture the `poolName` in a field of the class.

This field can then be asserted.

Reply:
Done